lib/puppet/application/preview.rb

```diff
@@ -324,12 +324,16 @@ def compile_diff(node, timestamp)
     @exit_code = 0
 
     begin
+      if Facter.value(:osfamily).nil?
```

Comment:
Is this really the way to get the facts for the node being compiled? I suspect this gets the fact on the machine running 'preview'.
The value we want is what got resolved when getting the node. I should be in a node parameter or something like that.